From: Nick Piggin <npiggin@suse.de>
Subject: x86 set cacheline size to 128 bytes for generic CPUs
Patch-mainline: no (could be submitted?)

GENERIC CPU selection includes Pentium 4 which effectively has 128
byte cachelines for false sharing purposes.

Index: linux-2.6.27/arch/x86/Kconfig.cpu
===================================================================
--- linux-2.6.27.orig/arch/x86/Kconfig.cpu
+++ linux-2.6.27/arch/x86/Kconfig.cpu
@@ -303,8 +303,8 @@ config X86_CPU
 # Define implied options from the CPU selection here
 config X86_L1_CACHE_BYTES
 	int
-	default "128" if MPSC
-	default "64" if GENERIC_CPU || MK8 || MCORE2 || MATOM || X86_32
+	default "128" if GENERIC_CPU || MPSC
+	default "64" if MK8 || MCORE2 || MATOM || X86_32
 
 config X86_INTERNODE_CACHE_BYTES
 	int
@@ -316,10 +316,10 @@ config X86_CMPXCHG
 
 config X86_L1_CACHE_SHIFT
 	int
-	default "7" if MPENTIUM4 || MPSC
+	default "7" if X86_GENERIC || GENERIC_CPU || MPENTIUM4 || MPSC
 	default "4" if X86_ELAN || M486 || M386 || MGEODEGX1
 	default "5" if MWINCHIP3D || MWINCHIPC6 || MCRUSOE || MEFFICEON || MCYRIXIII || MK6 || MPENTIUMIII || MPENTIUMII || M686 || M586MMX || M586TSC || M586 || MVIAC3_2 || MGEODE_LX
-	default "6" if MK7 || MK8 || MPENTIUMM || MCORE2 || MATOM || MVIAC7 || X86_GENERIC || GENERIC_CPU
+	default "6" if MK7 || MK8 || MPENTIUMM || MCORE2 || MATOM || MVIAC7
 
 config X86_XADD
 	def_bool y
